pkg/reconciler/common/prune.go

```diff
@@ -266,7 +265,12 @@ func (pruner *Pruner) createJobContainers(nsConfig *pruneConfigPerNS, ns string)
 	return containers
 }
 
-func (pruner *Pruner) checkAndCreate(ctx context.Context, uniquePruneNs, schedule string, pruneContainers []corev1.Container, listOpt string, configChanged bool) error {
+func (pruner *Pruner) checkAndCreate(ctx context.Context, uniquePruneNs, schedule string, pruneContainers []corev1.Container, listOpt string, configChanged bool, tC *v1alpha1.TektonConfig) error {
+	// calculate tektonConfig hash
+	computeTcConfigHash, err := hash.Compute(tC.Spec.Config)
```

Comment:
Should we change `computeTcConfigHash` to `nodeselectorConfigHash`, as we are computing the hash for the NodeSelector and Tolerations .

Reply:
`computeTcConfigSpecConfigHash` how does this sound ? I know its little confusing but we are computing tekton's config spec.config's hash 
```
config:
    nodeSelector:
     node-role.kubernetes.io/master: ""
    tolerations:
    - key: node-role.kubernetes.io/master
      effect: NoSchedule
      operator: Exists
```

Reply:
changed to nodeselectorConfigHash

Reply:
@piyush-garg , as per my understanding `tektonconfig.spec.config` can only contain nodeselector and tolerations @pradeepitm12 is my assumption correct ?

Reply:
Compute function computes the hash of config struct irrespective of the fields inside that

Reply:
That is true, anyways I have changed it to calculate the hash of nodeSelector

Reply:
Why we are depending on only nodeSelector, prune Jobs should change based on tolerations also